gas-oracle/tokenprice/tokenprice.go

```diff
@@ -9,10 +9,48 @@ import (
 	"github.com/go-resty/resty/v2"
 )
 
-var errHTTPError = errors.New("http error")
+type TokenRatioMode uint64
 
-// NewClient create a new Client given a remote HTTP url and update frequency
-func NewClient(url string, frequency uint64) *Client {
+// Client is an HTTP based TokenPriceClient
+type Client struct {
+	client              *resty.Client
+	uniswapQuoterClient *uniswapClient
+	frequency           time.Duration
+	lastRatio           float64
+	lastEthPrice        float64
+	lastUpdate          time.Time
+	tokenRatioMode      TokenRatioMode
+}
+
+type TokenPrice struct {
+	Symbol string `json:"symbol"`
+	Price  string `json:"price"`
+}
+
+type Result struct {
+	RetCode int
+	Result  TokenPrice
+}
+
+var (
+	errHTTPError = errors.New("http error")
+
+	// DefaultTokenRatio is eth_price / mnt_price
+	DefaultTokenRatio = 4000
+	// DefaultETHPrice is default eth_price
+	// If SwitchOneDollarTokenRatio valid, use DefaultETHPrice to set token_ratio to mnt price is 1$
+	DefaultETHPrice = 1800
+
+	// RealTokenRatioMode use eth_price / mnt_price to set token_ratio
+	RealTokenRatioMode = TokenRatioMode(0)
+	// DefaultTokenRatioMode use DefaultTokenRatio to set token_ratio
+	DefaultTokenRatioMode = TokenRatioMode(1)
+	// OneDollarTokenRatioMode use eth_price to set token_ratio, so mnt price is 1$
+	OneDollarTokenRatioMode = TokenRatioMode(2)
```

Comment:
Can it be adjusted to arrange in order of priority using mod?

Reply:
Modified, priority will show like this:
- RealTokenRatioMode = TokenRatioMode(0)
- OneDollarTokenRatioMode = TokenRatioMode(1)
- DefaultTokenRatioMode = TokenRatioMode(2)